src/runtime-rs/crates/runtimes/virt_container/src/sandbox.rs

```diff
@@ -242,28 +244,23 @@ impl VirtSandbox {
         Ok(())
     }
 
-    async fn prepare_rootfs_config(&self) -> Result<BlockConfig> {
+    async fn prepare_rootfs_config(&self) -> Result<Option<BlockConfig>> {
         let boot_info = self.hypervisor.hypervisor_config().await.boot_info;
 
-        let image = {
-            let initrd_path = boot_info.initrd.clone();
-            let image_path = boot_info.image;
-            if !initrd_path.is_empty() {
-                Ok(initrd_path)
-            } else if !image_path.is_empty() {
-                Ok(image_path)
-            } else {
-                Err(anyhow!("failed to get image"))
-            }
+        if !boot_info.initrd.is_empty() {
+            return Ok(None);
```

Comment:
This looks like hypervisor drivers will stop being called to add a block device whose `path_on_host` points in fact to an initrd, not image.  If so then that's great!

However, are there any other such consequences that could break a hypervisor support?  This is common code which affects everyone.

Reply:
Yeah I had to add the check and will be happy to get rid of it again since it's somewhat ugly. :-)